datalake/src/main/java/com/sequenceiq/datalake/flow/detach/SdxDetachEvent.java

```diff
@@ -1,17 +1,20 @@
 package com.sequenceiq.datalake.flow.detach;
 
 import com.sequenceiq.datalake.flow.detach.event.SdxDetachFailedEvent;
-import com.sequenceiq.datalake.flow.detach.event.SdxDetachInprogressEvent;
-import com.sequenceiq.datalake.flow.detach.event.SdxDetachSuccessEvent;
 import com.sequenceiq.flow.core.FlowEvent;
 
 public enum SdxDetachEvent implements FlowEvent {
```

Comment:
the pain is coming from the fact that this class uses 
```SdxDetachEvent(Class eventClass) {
        this.event = eventClass.getSimpleName();
        this.event = eventClass.getSimpleName();
    }
    }
```
Instead of `EventSelectorUtil` which is the generic selector implementation for events...

Reply:
ah messed up the copy, but it's from the constructor code

Reply:
Sorry could you expand on this a little? I am seeing inconsistent use of different logic throughout different events. The Start and Stop events use the same as what we are using here, and a lot of other events just use basic strings within the enum objects. 
Also that constructor is essentially not used anywhere so I am not sure how important it is to get it using something else.

I am open to suggestions though. This is my first flow change so I am new to a lot of this.

Reply:
sure, let me elaborate a bit

the problem with this approach this case have:
- it has a selector as a string in the event itself (`SdxDetachFailedEvent`)
- `SdxDetachEvent` calculates the selector based on a unique logic based on the class
If you rename the class but you forget about the string based selector, then this would only be recognized during runtime, as there is no IDE or compile time support.

Suggested way to handle selectors and event:
- for class based selectors use the `EventSelectorUtil`, and use it everywhere to ensure unity. So if you rename the class using the IDE it would rename it everywhere or it would cause a compile time error
- if you need string based selector, like you are reusing `SdxEvent` as you don't need anything else, then in this case in the `SdxDetachEvent` enum you would define an event with the string value, and when you create the event and set the selector you should use `SdxDetachEvent#getEvent`. This would ensure if the string changes for any reason it would still work

So the purpose is to use the same and proofed methods, and depend on the IDE and compiler as much as possible to ensure no errors are introduced if a change happens. I hope this clarifies, if not please reach out to me.

Reply:
Made changes to the `SdxDetachEvent` and `SdxDetachFailedEvent` classes. Thank you for the thorough information it will help me out a lot in the future.